pie/storage/__init__.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+
+from pydoc import locate
+from typing import Any
+
+from nextcord.ext import commands
+
+from pie.storage.database import StorageData
+
+
+def get(module: commands.Cog, guild_id: int, key: str, default_value=None) -> Any:
+    """Get data from persistant DataStorage base on module and guild.
+    For saving global data (non-guild related) set guild_id to 0.
+
+    If value is not found in DB or it's type is unknown,
+    it returns default_value, which if not set is None.
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+        default_value: This argument is returned if value is not found in database
+
+    Returns:
+        Any: value stored in DB
+
+    Raises:
+        ValueError: Raised if value type change fails
+    """
+
+    db_value = StorageData.get(module.qualified_name, guild_id, key)
+    if not db_value:
+        return default_value
+
+    t = locate(db_value.type)
+
+    if not t:
+        return default_value
+
+    value = t(db_value.value)
+
+    return value
+
+
+def exists(module: commands.Cog, guild_id: int, key: str) -> bool:
+    """Checks if data for module, key and guild_id combination
+    are present in DB.
```

Comment:
I'm not entirely convinced this is needed, as per previous discussion. Why would you be saving `None` to the database, if `None` is the default value that can be implied by the presence of the item.

Reply:
Do we want to support setting the value to `None`? I'm not sure if it shouldn't be enough to use the absence of value as `None`. I may not be seeing the value of storing `None`.

Reply:
Here's example:

You want to set some limit for something. There's tested and optimal default value of 200. But if bot owner / guild owner wants to remove that limit, he can just set it to None. Next time when bot boots up, this value is not initialized.

If you want to do this without setting to None, you would have to set it to some ridicoulously high value, which is in my opinion not good, the intentions are not that clean.